src/main/kotlin/com/github/suusan2go/kotlinfillclass/inspections/BaseFillClassInspection.kt

```diff
@@ -119,30 +146,87 @@ open class FillClassFix(
 
     override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
         val argumentList = descriptor.psiElement as? KtValueArgumentList ?: return
-        val (resolvedCall, functionDescriptor) = (argumentList.parent as? KtCallElement)?.analyze() ?: return
-        argumentList.fillArguments(functionDescriptor.valueParameters, resolvedCall)
+        val call = argumentList.parent as? KtCallElement ?: return
+        val descriptors = analyze(call).ifEmpty { return }
+
+        val lambdaArgument = call.lambdaArguments.singleOrNull()
+        val editor = argumentList.findExistingEditor()
+        if (descriptors.size == 1 || editor == null) {
+            argumentList.fillArguments(descriptors.first().second, editor, lambdaArgument)
+        } else {
+            val listPopup = createListPopup(argumentList, descriptors, editor)
+            JBPopupFactory.getInstance().createListPopup(listPopup).showInBestPositionFor(editor)
+        }
+    }
+
+    private fun createListPopup(
+        argumentList: KtValueArgumentList,
+        descriptors: List<Pair<KtFunction, FunctionDescriptor>>,
+        editor: Editor?,
+    ): BaseListPopupStep<String> {
+        val functionName = descriptors.first().let { (_, descriptor) ->
+            if (descriptor is ClassConstructorDescriptor) {
+                descriptor.containingDeclaration.name.asString()
+            } else {
+                descriptor.name.asString()
+            }
+        }
+        val functions = descriptors
+            .sortedBy { (_, descriptor) -> descriptor.valueParameters.size }
+            .associate { (function, descriptor) ->
+                val key = function.valueParameters.joinToString(
+                    separator = ", ",
+                    prefix = "$functionName(",
+                    postfix = ")",
+                    transform = { "${it.name}: ${it.typeReference?.text ?: ""}" },
+                )
+                key to descriptor
+            }
+        return object : BaseListPopupStep<String>("Choose Function", functions.keys.toList()) {
+            override fun isAutoSelectionEnabled() = false
+
+            override fun onChosen(selectedValue: String, finalChoice: Boolean): PopupStep<*>? {
+                if (finalChoice) {
+                    val parameters = functions[selectedValue]?.valueParameters.orEmpty()
+                    CommandProcessor.getInstance().runUndoTransparentAction {
+                        runWriteAction {
+                            argumentList.fillArguments(parameters, editor)
```

Comment:
I'm having the same problem as #85 because of missing `lambdaArgument`.

Could you add `lambdaArgument` argument to `createListPopup` ?

Reply:
[12d1298](https://github.com/suusan2go/kotlin-fill-class/pull/102/commits/12d1298401e9c949578281d67b1cb622663e37f9)